### PR TITLE
Jabenk/physics dev

### PR DESCRIPTION
**Pull request by jabenk**
Adding a low-frequency server transform upload stream. The physics bridge updates are not readable for the server.
For late joining clients it is important to have the current positions on the server, this should not happen based on subscriptions.
https://github.com/microsoft/mixed-reality-extension-sdk/issues/637

### CODE REVIEW

File: MRETestBed/Assets/Scenes/SynchronizationTest-localhost-3clients.unity [hunk 1]
```diff
@@ -0,0 +1,964 @@
+%YAML 1.1
+%TAG !u! tag:unity3d.com,2011:
```
Comment (nik0la31) [inline MRETestBed/Assets/Scenes/SynchronizationTest-localhost-3clients.unity:2]: why are you adding this file?

File: MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs [hunk 2]
```diff
@@ -327,6 +327,21 @@ private void SendPhysicsUpdate()
...
 			}
+
+			//low frequency server upload transform stream
+			{
+				float systemTime = UnityEngine.Time.time;
+				if (PhysicsBridge.shouldSendLowFrequencyTransformUpload(systemTime))
+				{
+					PhysicsTranformServerUploadPatch serverUploadPatch =
+						PhysicsBridge.GenerateServerTransformUploadPatch(InstanceId, systemTime);
+					// upload only if there is a real difference
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs:338]: >is a real difference [](start = 29, length = 20)

if there are upates

File: MREUnityRuntime/MREUnityRuntimeLib/Messaging/Payloads/NetworkCommandPayloads.cs [hunk 3]
```diff
@@ -490,6 +490,18 @@ public class PhysicsBridgeUpdate : NetworkCommandPayload
 		/// <summary>
 		/// Physics bridge update with transforms for owned rigid bodies.
 		/// </summary>
-		public PhysicsBridgePatch PhysicsBridge { get; set; }
+		public PhysicsBridgePatch PhysicsBridgePatchContent { get; set; }
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Messaging/Payloads/NetworkCommandPayloads.cs:493]: >PhysicsBridgePatchContent [](start = 28, length = 25)

maybe just 'Patch'

File: MREUnityRuntime/MREUnityRuntimeLib/Messaging/Payloads/NetworkCommandPayloads.cs [hunk 4]
```diff
@@ -490,6 +490,18 @@ public class PhysicsBridgeUpdate : NetworkCommandPayload
...
+	/// <summary>
+	/// Unidirectional
+	/// low frequency to upload the positions to the server.
+	/// </summary>
+	public class PhysicsTranformServerUpload : NetworkCommandPayload
+	{
+		/// <summary>
+		/// Physics bridge updates transforms with low frequency on the server.
+		/// </summary>
+		public PhysicsTranformServerUploadPatch PhysicsTranformServer { get; set; }
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Messaging/Payloads/NetworkCommandPayloads.cs:505]: >PhysicsTranformServerUploadPatch [](start = 9, length = 32)

maybe just 'Patch'

File: MREUnityRuntime/MREUnityRuntimeLib/Messaging/Events/Types/PhysicsBridgeUpdated.cs [hunk 5]
```diff
@@ -27,8 +26,33 @@ internal override void SendEvent(MixedRealityExtensionApp app)
...
+		internal override void SendEvent(MixedRealityExtensionApp app)
+		{
+			if (app.Protocol == null)
+			{
+				return;
+			}
+
+			app.Protocol.Send(new PhysicsTranformServerUpload()
+			{
+				PhysicsTranformServer = _physicsTransformUploadPatchPatch
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Messaging/Events/Types/PhysicsBridgeUpdated.cs:53]: >PatchPatch [](start = 51, length = 10)

PatchPatch?

File: MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs [hunk 6]
```diff
@@ -447,5 +468,88 @@ public void LateUpdate()
...
+			numownedbodies = 0;
+			foreach (var update in allUpdates)
+			{
+				// add the updates
+				ret.updates[numownedbodies++] = update;
+			}
+			// store the last time
+			_lastServerTransformUploadSentTime = systemTime;
+
+			//
```
Comment (nik0la31) [inline MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs:551]: >//  [](start = 3, length = 3)

delete comment

File: MRETestBed/Assets/Scenes/SynchronizationTest-localhost-3clients.unity (hunk 1 above)
Comment (jabenk) [reply MRETestBed/Assets/Scenes/SynchronizationTest-localhost-3clients.unity:2]: I see some weird behaviors with 3 clients in some tests, and here I can also test the late joining of the 3rd client.

**Pull request merged**